src/_posts/2020-07-18-packaging-with-packer.md

```diff
@@ -0,0 +1,601 @@
+---
+layout: post
+asset-type: post
+title: Packaging with Packer
+date: 2020-07-18 00:00:00 +00:00
+author: André Guelfi Torres
+description: What is Packer? How it can help us and how to use it.
+image:
+    src: 
+    attribution: 
+        text: 
+        href: 
+tags:
+    - packer
+    - devops
+    - infrastructure as code
+    - automation
+---
+
+# AMIs and the Old Way
+
+We all know how important is Infrastructure as Code(IaC) in a DevOps environment, the main reasons are that code is:
+
+- Reproducible: We can run how many times we want and we are going to get the expected result, unlike manual changes automation won't commit silly mistakes of forgetting to run a command.
+- Versionable: We can version using git, track the changes and be able to use the same collaboration tools and techniques like Merge Requests and Review.
+- Automated: Besides triggering you don't have to do anything to get the infrastructure up and running.
+
+In case your project isn't following the latest trend and using something like serverless or Kubernetes there's a big chance that you are using Virtual Machines, even if you are in the cloud. In this case, there are two kinds of tools for IaC in VM environments. 
+
+- Provisioning Tools: Those are the ones to create the infrastructure, like Terraform or CloudFormation.
+- Configuration Management Tools: Which are the ones used to setup machines, deploy applications and configure them. We have many tools for that like Ansible, Chef, Puppet.
+
+One of the issues that we have is time, after provisioning everything we have to install everything to the box, usually, it's going to be multiple applications and runtimes to install beofre having anything running. During this process of installation something might fail due a dependency that isn't available anymore, like a repository for apt that is missing or even a third party that is having connectivity issues. 
+
+Those applications/runtimes are less prone to change than your configurations or your applications. We can use pre-baked images but there's a problem with that, the creation of pre-baked images is very manual, the process goes by:
```

Comment:
Update sentence:

Those applications or runtimes are less prone to change than your configurations or your applications. We can use pre-baked images, but there's a problem with that, the creation of pre-baked images is very manual. The process goes by:

(Elaborate on the process goes by. Suggestion: The process for pre-baked images is complete by...)